src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -321,91 +326,99 @@ int MPIDI_OFI_get_huge_event(struct fi_cq_tagged_entry *wc, MPIR_Request * req)
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_GET_HUGE_EVENT);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_GET_HUGE_EVENT);
 
-    if (recv_elem->localreq && recv_elem->cur_offset != 0) {    /* If this is true, then the message has a posted
-                                                                 * receive already and we'll be able to find the
-                                                                 * struct describing the transfer. */
-        if (MPIDI_OFI_COMM(recv_elem->comm_ptr).enable_striping) {
-            /* Subtract one stripe_chunk_size because we send the first chunk via a regular message
-             * instead of the memory region */
-            recv_elem->stripe_size = (recv_elem->remote_info.msgsize - MPIDI_OFI_STRIPE_CHUNK_SIZE)
-                / MPIDI_OFI_global.num_nics;    /* striping */
-
-            if (recv_elem->stripe_size > MPIDI_OFI_global.max_msg_size) {
-                recv_elem->stripe_size = MPIDI_OFI_global.max_msg_size;
-            }
-            if (recv_elem->chunks_outstanding)
-                recv_elem->chunks_outstanding--;
-            bytesLeft = recv_elem->remote_info.msgsize - recv_elem->cur_offset;
-            bytesToGet = (bytesLeft <= recv_elem->stripe_size) ? bytesLeft : recv_elem->stripe_size;
-        } else {
-            /* Subtract one max_msg_size because we send the first chunk via a regular message
-             * instead of the memory region */
-            bytesLeft = recv_elem->remote_info.msgsize - recv_elem->cur_offset;
-            bytesToGet = (bytesLeft <= MPIDI_OFI_global.max_msg_size) ?
-                bytesLeft : MPIDI_OFI_global.max_msg_size;
+    /* Note: this function is invoked from three places --
+     * 1. In recv_huge_event, when recv buffer is matched and first chunk received, and
+     *    when control message (with remote info) has also been received.
+     * 2. In MPIDI_OFI_get_huge, as a callback when control message is received, and
+     *    when first chunk has been matched and received.
+     *
+     * recv_huge_event will fill the local request information, and the control message
+     * callback will fill the remote (sender) information. Lastly --
+     *
+     * 3. As the event function when RDMA read (issued here) completes.
+     */
```

Comment:
I meant that it would be nice to have this before the function is defined (i.e., line 319). I don't care enough to hold things up though.

Reply:
So without multi-NIC striping each read is done one at a time? I don't know this code extremely well, so just for my understanding. I think the changes here are fine.

Reply:
Yeah, though it won't divide up the message into chunks so it will read up to the maximum message size (usually 2GB) and get the rest of message in a single go.

Reply:
I see, Didn't realize `max_msg_size` is huge, so I guess that's not an issue :)

Reply:
> 
> 
> I meant that it would be nice to have this before the function is defined (i.e., line 319). I don't care enough to hold things up though.

The comment block is moved to where you liked :)